tls/s2n_next_protocol.h

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#pragma once
+
+#include "utils/s2n_result.h"
+
+S2N_RESULT s2n_calculate_padding(uint8_t protocol_len, uint8_t *padding_len);
+int s2n_write_npn_protocol(struct s2n_connection *conn, struct s2n_stuffer *out);
+int s2n_read_npn_protocol(struct s2n_connection *conn, struct s2n_stuffer *extension);
```

Comment:
Why do these functions need to be in this header? It tests need them, couldn't you just put them in the test file?